src/ActionMonitor/ActionMonitor.php

```diff
@@ -79,6 +79,34 @@ class_exists( 'WPGraphQL' ) && method_exists( 'WPGraphQL', 'debug' )
 		// Trigger webhook dispatch
 		add_action( 'shutdown', [ $this, 'trigger_dispatch' ] );
 
+		// allow any role to use Gatsby Preview
+		add_action( 'admin_init', [ $this, 'action_monitor_add_role_caps' ], 999 );
+	}
+
+	/**
+	 * For Action Monitor, all of these roles need to be able to view and edit private action monitor posts so that Preview works for all roles.
+	 */
+	public function action_monitor_add_role_caps() {
+		$doing_graphql_request
+					= defined( 'GRAPHQL_REQUEST' ) && true === GRAPHQL_REQUEST;
+
+		if ( $doing_graphql_request ) {
+			// we only need to add roles one time. checking capabilities repeatedly isn't needed, just when the user is in the admin area is fine.
+			return;
+		}
+
+		$roles = array( 'editor', 'administrator', 'contributor', 'author' );
```

Comment:
This means all roles can view all action monitor items, correct? Would there ever be a site that wouldn't want this ability by default?

Reply:
It means all these roles can view _private_ action monitor posts (they can already view action monitor posts). Most action monitor posts are public except for those related to previews. There isn't any sensitive info in there though - it just doesn't make sense for preview action monitor posts to be public since you can only utilize them when authenticated.
This also leaves out the `subscriber` role since they can't view/edit/preview anything anyway. 
You're right though that perhaps some sites might not want this under some circumstances. I'll add a filter here so it can be changed.